Modules/Loadable/VolumeRendering/MRML/vtkMRMLShaderPropertyNode.h

```diff
@@ -0,0 +1,88 @@
+/// vtkMRMLShaderPropertyNode - MRML node to represent custom volume shader
+/// variables and replacement code
+
+#ifndef __vtkMRMLShaderPropertyNode_h
+#define __vtkMRMLShaderPropertyNode_h
+
+// VolumeRendering includes
+#include "vtkSlicerVolumeRenderingModuleMRMLExport.h"
+
+// MRML includes
+#include "vtkMRMLStorableNode.h"
+
+// VTK includes
+class vtkShaderProperty;
+class vtkUniforms;
+
+/// \brief vtkMRMLShaderPropertyNode volume shader custom code and
+/// custom uniform variables defined by users or specialized rendering
+/// modules.
+class VTK_SLICER_VOLUMERENDERING_MODULE_MRML_EXPORT vtkMRMLShaderPropertyNode
+  : public vtkMRMLStorableNode
+{
+public:
+
+  /// Create a new vtkMRMLShaderPropertyNode
+  static vtkMRMLShaderPropertyNode *New();
+  vtkTypeMacro(vtkMRMLShaderPropertyNode,vtkMRMLStorableNode);
+  void PrintSelf(ostream& os, vtkIndent indent) VTK_OVERRIDE;
+
+  /// Don't change its scalarOpacity, gradientOpacity or color on the volume property
+  /// but use the methods below. It wouldn't observe them.
+  vtkGetObjectMacro(ShaderProperty, vtkShaderProperty);
+
+  /// Get the list of user-defined uniform variables.
+  vtkUniforms * GetVertexUniforms();
+  vtkUniforms * GetFragmentUniforms();
+
+  //--------------------------------------------------------------------------
+  /// MRMLNode methods
```

Comment:
Doxy prefix only for the briefs for single functions, please. This way I think the doxy for CreateNodeInstance will say "MRMLNode methods", which is neither helpful nor true.

Reply:
fixed.